src/worker/health.rs

```diff
@@ -22,6 +24,10 @@ where
     ///   but should _continue_ processing its current job (if any);
     ///
     /// See more details [here](https://github.com/contribsys/faktory/blob/b4a93227a3323ab4b1365b0c37c2fac4f9588cc8/server/workers.go#L13-L49).
+    ///
+    /// Note that this method is not cancellation safe. We are using an interval timer internally, that
```

Comment:
This is a good doc addition. Makes me wonder: do we call this method in the context of a `select!` ever at the moment? Or call it from an `async fn` that a user may `select!` over (and that we thus need to also document isn't cancellation safe)?

Reply:
Not just yet - this comment belongs to #57 where we do call in within a `select!`. Moved it there.

 As for the `Worker::run` (and `Worker::run_to_completion`) those are actually also not cancellation safe because of the `AsyncWriteExt::write_all`. It is also not safe to run the `Worker::run` in a loop due to the terminated worker scenario: if they try to re-run a terminated worker, we will [panic](https://github.com/jonhoo/faktory-rs/blob/3c966bdca641f4fc101d88b0098f6c99efbf7210/src/worker/mod.rs#L352-L355) right away.